client/jetpack-cloud/sections/auth/index.ts

```diff
@@ -6,14 +6,15 @@ import page from 'page';
 /**
  * Internal dependencies
  */
-import { authConnectPath, authTokenRedirectPath } from './paths';
-import { connect, tokenRedirect } from './controller';
+import { authConnectPath, authTokenRedirectPath, authUserInitializeRedirectPath } from './paths';
+import { connect, tokenRedirect, userInitialize } from './controller';
 import { makeLayout, render as clientRender } from 'calypso/controller';
 import config from 'calypso/config';
 
-export default () => {
+export default (): void => {
 	if ( config.isEnabled( 'jetpack-cloud' ) ) {
 		page( authConnectPath(), connect, makeLayout, clientRender );
-		page( authTokenRedirectPath(), tokenRedirect, makeLayout, clientRender );
+		page( authTokenRedirectPath(), tokenRedirect, userInitialize, makeLayout, clientRender );
+		page( authUserInitializeRedirectPath(), userInitialize, makeLayout, clientRender );
```

Comment:
With this new OAuth form, we no longer need to redirect users to the `tokenRedirect` controller so we make them skip the step and redirect them directly to the user initialization (fetching).

Reply:
For context, the `tokenRedirect` controller is in charge of extracting an access token from the URL and storing it on local storage. This is no longer necessary because, in this new context, the token is obtained from the response of the login request.